AstSemantics.md

```diff
@@ -96,107 +120,59 @@ following rules:
 Wrapping of integers simply discards any upper bits; i.e. wrapping does not
 perform saturation, trap on overflow, etc.
 
-## Addressing local variables
+In addition to storing a value to linear memory, store instructions also return
+their value operand, with no conversion applied.
 
-Each function has a fixed, pre-declared number of local variables which occupy a single
-index space local to the function. Parameters are addressed as local variables. Local
-variables do not have addresses and are not aliased in the globals or memory. Local
-variables have *Local types* and are initialized to the appropriate zero value for their
-type at the beginning of the function, except parameters which are initialized to the values
-of the arguments passed to the function.
+### Addressing
 
-  * `get_local`: read the current value of a local variable
-  * `set_local`: set the current value of a local variable
-
-The details of index space for local variables and their types will be further clarified,
-e.g. whether locals with type `int32` and `int64` must be contiguous and separate from 
-others, etc.
+Each linear memory access operation also has an address operand and an immediate
+integer offset attribute. The infinite-precision sum of the address operand's
+value with the offset attribute's value is called the *effective address*.
 
-## Control flow structures
+Effective addresses are interpreted as unsigned byte indices into the linear
+memory storage, starting at `0`. Linear memory accesses access memory starting
+at this byte and extending for the number of bytes implied by the memory
+attribute.
```

Comment:
I find this slightly confusing: is it `base_address_offset_from_0_in_byte + offset_in_bytes` or is the offset instead then number of elements to index from (so, multiply by `sizerof(type)` for number of bytes)?